cns/restserver/ipam.go

```diff
@@ -0,0 +1,253 @@
+// Copyright 2017 Microsoft. All rights reserved.
+// MIT License
+
+package restserver
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/Azure/azure-container-networking/cns"
+	"github.com/Azure/azure-container-networking/cns/logger"
+)
+
+func newIPConfig(ipAddress string, prefixLength uint8) cns.IPSubnet {
+	return cns.IPSubnet{
+		IPAddress:    ipAddress,
+		PrefixLength: prefixLength,
+	}
+}
+
+func newPodState(ipaddress string, prefixLength uint8, id, ncid, state string) cns.ContainerIPConfigState {
```

Comment:
Export this, request controller needs this when populating CNS with available ipconfigs